targets technote

wlandau opened the pull request:
As [proposed here](https://github.com/ropensci/software-review/issues/401#issuecomment-758859285), I would like to contribute a technote about `targets` (for 2021-02-03). 

* [x] I have read the Content Guidelines.
* [x] I have read the Technical Guidelines.
* [x] I used or followed the R Markdown or Markdown template.
* [x] I have followed the Style Guide.
* [x] I created or updated my author metadata with correct folder name.
* [x] I have added relevant tags after browsing existing tags (incuding "community" tag).
* [x] I ran `roblog::ro_lint_md()` on index.md (optional).
* [x] I ran `roblog::ro_check_urls()` on index.md (optional).
* [x] I ran a spell-check on index.md.
* [x] I have added the tags - Software Peer Review, my-packagename.
* [x] I have added the package-version YAML tag.
* [x] I have added acknowledgement of the reviewers' work (with links to reviewers).
* [x] I have added a link to the software peer review thread.

wlandau commented:
I believe the text itself is ready for review, but the [preview](https://deploy-preview-122--ropensci.netlify.app/technotes/2021/02/03/targets/) is not quite rendering right. I will add comments inline.

wlandau commented on content/technotes/2021-02-03-targets.Rmd:
```diff
@@ -0,0 +1,434 @@
+## What is `targets`?
+
+<!--html_preserve-->
+{{< figure src = "technotes/2021/02/03/targets.png" alt = "targets hex logo" >}}
```
I am having trouble with static images like these. They do not seem to render.

wlandau replied:
I inspected the rendered source and found the paths I needed.

wlandau commented on content/technotes/2021-02-03-targets.Rmd:
```diff
@@ -0,0 +1,434 @@
+<!--/html_preserve-->
+
+{{< imgtxt >}}
+[`targets`](https://docs.rOpenSci.org/targets/)[^2] package is a new pipeline toolkit for R.
```
Footnotes inside `{{< imgtext >}}` do not seem to render in the preview.

wlandau replied:
I decided to avoid `{{< imgtext >}}`.

wlandau commented on content/technotes/2021-02-03-targets.Rmd:
````diff
@@ -0,0 +1,434 @@
+```{r minimal-graph, hugoopts = list(alt = "minimal graph"), eval = TRUE}
+# R console
+library(targets)
+tar_visnetwork()
````
This chunk should show a [`visNetwork`](https://datastorm-open.github.io/visNetwork/) HTML widget, but the raw HTML is inserted verbatim.

wlandau replied:
I mostly got this to work using external HTML files, but the widths and heights were difficult to adjust, and the legend is still off the page. In future posts, it would be nice to just embed the widgets directly like I usually do for R Markdown. I posted an issue at https://github.com/ropensci-org/blog-guidance/issues/168.

wlandau commented on content/technotes/2021-02-03-targets.Rmd:
````diff
@@ -0,0 +1,434 @@
+
+```{r histogram, hugoopts = list(alt = "histogram")}
+# R console
+tar_read(hist) # see also tar_load()
````
The static image generated by this code chunk does not render.

wlandau replied:
I worked around this issue with a static image. From the guide, I think it is supposed to work and I am just missing something.

wlandau commented:
The netlify preview really helped me make changes, and I now believe the draft is ready for review.

stefaniebutland commented:
Thank you @wlandau! My colleague @steffilazerte will review the post. Publication on 2021-02-03 is confirmed

steffilazerte commented:
Awesome! I'll review soon so we get this ready for next Wednesday!

steffilazerte commented on content/technotes/2021-02-03-targets.Rmd:
```diff
@@ -0,0 +1,477 @@
+tags:
+  - Software Peer Review
+  - packages
+  - R
```
We're leaving off the 'R' tag for now
```suggestion
```

steffilazerte reviewed (commented):
Hi @wlandau,

Wonderful post! I heard so much about drake I really wanted to give it a try, but when I heard targets was in the works, I thought I should wait. Now I'm super busy but also so desperate to give targets a try! I compile a lot of workshop presentations and feel like targets might be a way of making my life so much easier!

I have one suggestion and one comment (a new record low for me). I was also struggling with the twitter card (see below), but can't see what could be the problem.

Thanks again!

* [X] post follows Content Guidelines
* [X] post follows Style Guide
* [X] title is in Title Case
* [X] publication date is ok
* [X] alternative text of images is informative
* [ ] Twitter metadata looks ok (paste post preview link in [Twitter card validator](https://cards-dev.twitter.com/validator); might have to click twice on "Preview card")
  - **Not sure what's going on here, the setup looks fine but I don't see the image in the validator. @wlandau, @stefaniebutland, @maelle? Is this just me?** 
* [X] author metadata is provided with correct folder name
* [X] html not included in pull request of Rmd post
* [X] I ran `roblog::ro_lint_md()` on index.md
* [X] I ran `roblog::ro_check_urls()` on index.md
* [X] I ran a spell-check on index.md
* [X] YAML subject tags are ok (add "community" for post by non-staff non-editor)
* [X] YAML package-version included
* [X] YAML subject tags - software peer review, packagename
* [X] acknowledges and links to reviewers
* [X] links to peer review thread

wlandau commented on content/technotes/2021-02-03-targets.Rmd:
```diff
@@ -0,0 +1,477 @@
+With pipelines, data science feels smoother and more rewarding, and the results are worthy of more trust.
+
+<!--html_preserve-->
+<!-- I thought about following the blog guide's recommendation to use the Hugo shortcode for this tweet, but I feel the media and emojis are a bit much. -->
```
Thanks! I am considering requesting a new shortcode that renders like this. I had to manually copy the rendered blockquote and remove the media.

steffilazerte replied:
Great idea!

wlandau commented:
Thanks, @steffilazerte! I made some more changes:

* Removed the R tag.
* Fixed the Twitter card. (Turns out I just had the wrong path.)
* Edited the description to be different from the title. The Twitter card looks better now.
* Created a new image with all the R Targetopia hexes tessellated. Now in both the post and the Twitter card.

steffilazerte commented:
Thanks @wlandau everything looks beautiful! I've opened an issue in the blog guide to figure out if the guidance for twitter card image links need to be updated (I'm pretty sure yes), but thanks for figuring out the problem!

Here's a draft tweet to announce your blog post, I'm not super twitter-creative, so feel free to suggest, comment or rewrite!

---

[technote] targets: Democratizing Reproducible Analysis Pipelines

@wmlandau presents R 📦 targets, long-term successor of drake!  🎉🎉🎉

A new pipeline tool, a new ecosystem, and how you can help

https://ropensci.org/technotes/2021/02/03/targets/

#rstats

(1/2)

---

🙏 Thank yous go out to 

- rOpenSci reviewers @limnoliver and @tjmahr
- Feedback from users, esp @mattwrkntn, @timingliu, @MilesMcBain, @gorkang, @brunotestaguzza, @noamross, Kendon Bell
- Richard Payne inspired target factories and R Targetopia
- All drake contributors!

(2/2)

wlandau commented:
Fantastic, Steffi! Would you thank Mauro (editor) in the tweet alongside the reviewers? Other than that, it looks great.

steffilazerte commented:
Absolutely! See the edits (also some shortening to squeeze into the character count)

wlandau commented:
Thanks. Super minor: could `drake` be lowercase?

Also, I think I will go back and try to add some interstitial whitespace [here](https://github.com/wlandau/roweb3/blob/targets/content/technotes/2021-02-03-targets/targetopia.png) and [here](https://github.com/wlandau/roweb3/blob/targets/content/technotes/2021-02-03-targets/card.png). In hindsight, the dark backgrounds blend together.

steffilazerte commented:
Totally valid! I would have corrected it too :)

wlandau commented:
Thanks! Really excited for when the post and tweet go out next week.

> Also, I think I will go back and try to add some interstitial whitespace here and here. In hindsight, the dark backgrounds blend together.

Done.

steffilazerte commented:
Beautiful! 🎉  💯
